### PR TITLE
added solution

### PR DESCRIPTION
**Pull request by Mzuha**

### CODE REVIEW

File: src/main/java/mate/jdbc/dao/ManufacturerDaoImpl.java [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+    private Manufacturer getManufacturer(ResultSet resultSet) {
+        try {
+            String name = resultSet.getString("name");
+            String country = resultSet.getString("country");
+            Long manufacturersId = resultSet.getObject("id", Long.class);
+            Manufacturer manufacturer = new Manufacturer(name, country);
+            manufacturer.setId(manufacturersId);
+            return manufacturer;
+        } catch (SQLException e) {
+            throw new RuntimeException("Cant get manufacturer from ResultSet ", e);
```
Comment (Veronika-Alekseienko) [inline src/main/java/mate/jdbc/dao/ManufacturerDaoImpl.java:122]:
```suggestion
            throw new RuntimeException("Cant get manufacturer from ResultSet " + resultSet, e);
```

File: src/main/java/mate/jdbc/dao/ManufacturerDaoImpl.java [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+    public boolean delete(Long id) {
+        String deleteByIdStatementSql =
+                "UPDATE manufacturers SET is_deleted = true WHERE id = ?;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement deleteByIdStatement =
+                        connection.prepareStatement(deleteByIdStatementSql)) {
+            deleteByIdStatement.setString(1, id + "");
+            return deleteByIdStatement.executeUpdate() >= 1;
+        } catch (SQLException e) {
+            throw new RuntimeException("Cant delete manufacturer by id from DB ", e);
```
Comment (Veronika-Alekseienko) [inline src/main/java/mate/jdbc/dao/ManufacturerDaoImpl.java:109]:
```suggestion
            throw new RuntimeException("Cant delete manufacturer by id from DB " + id, e);
```

File: src/main/java/mate/jdbc/dao/ManufacturerDaoImpl.java [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+            updateStatement.setString(1, manufacturer.getName());
+            updateStatement.setString(2, manufacturer.getCountry());
+            updateStatement.setString(3, manufacturer.getId() + "");
+            updateStatement.executeUpdate();
+            ResultSet generatedKeys = updateStatement.getGeneratedKeys();
+            if (generatedKeys.next()) {
+                Long id = generatedKeys.getObject("id", Long.class);
+                manufacturer.setId(id);
+            }
+
```
Comment (Veronika-Alekseienko) [inline src/main/java/mate/jdbc/dao/ManufacturerDaoImpl.java:92]: empty line

File: src/main/java/mate/jdbc/dao/ManufacturerDaoImpl.java [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+                PreparedStatement getByIdStatement
+                        = connection.prepareStatement(getByIdStatementSql)) {
+            getByIdStatement.setString(1, id + "");
+            ResultSet resultSet = getByIdStatement.executeQuery();
+            if (resultSet.next()) {
+                return Optional.of(getManufacturer(resultSet));
+            }
+            return Optional.empty();
+        } catch (SQLException e) {
+            throw new RuntimeException("Cant get all manufacturers from DB ", e);
```
Comment (Veronika-Alekseienko) [inline src/main/java/mate/jdbc/dao/ManufacturerDaoImpl.java:53]:
```suggestion
            throw new RuntimeException("Cant get all manufacturers from DB " +  id, e);
```

File: src/main/java/mate/jdbc/dao/ManufacturerDaoImpl.java [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+            createStatement.setString(1, manufacturer.getName());
+            createStatement.setString(2, manufacturer.getCountry());
+            createStatement.executeUpdate();
+            ResultSet generatedKeys = createStatement.getGeneratedKeys();
+            if (generatedKeys.next()) {
+                Long id = generatedKeys.getObject(1, Long.class);
+                manufacturer.setId(id);
+            }
+        } catch (SQLException e) {
+            throw new RuntimeException("Cant insert manufacturer to DB ", e);
```
Comment (Veronika-Alekseienko) [inline src/main/java/mate/jdbc/dao/ManufacturerDaoImpl.java:34]:
```suggestion
            throw new RuntimeException("Cant insert manufacturer to DB " + manufacturer, e);
```

File: src/main/java/mate/jdbc/dao/ManufacturerDao.java [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+    Manufacturer create(Manufacturer manufacturer);
+
+    Optional<Manufacturer> get(Long id);
+
+    List<Manufacturer> getAll();
+
+    Manufacturer update(Manufacturer manufacturer);
+
+    boolean delete(Long id);
+
```
Comment (Veronika-Alekseienko) [inline src/main/java/mate/jdbc/dao/ManufacturerDao.java:17]: empty line

File: src/main/java/mate/jdbc/Main.java [hunk 7]
```diff
@@ -1,7 +1,17 @@
...
+import mate.jdbc.models.Manufacturer;
+
 public class Main {
+
+    private static final Injector injector = Injector.getInstance("mate.jdbc");
+
     public static void main(String[] args) {
+        ManufacturerDao manufacturerDao =
+                (ManufacturerDao) injector.getInstance(ManufacturerDao.class);
+        Manufacturer manufacturer1 = new Manufacturer("123", "USA");
```
Comment (Veronika-Alekseienko) [inline src/main/java/mate/jdbc/Main.java:15]: empty line

File: src/main/java/mate/jdbc/Main.java [hunk 8]
```diff
@@ -1,7 +1,17 @@
...
```
Comment (Veronika-Alekseienko) [inline src/main/java/mate/jdbc/Main.java:14]: let's test your methods in main

**Review by Veronika-Alekseienko — CHANGES_REQUESTED** [review]
Good job!)

**Review by Veronika-Alekseienko — APPROVED** [review]

File: src/main/java/mate/jdbc/dao/ManufacturerDaoImpl.java [hunk 9]
```diff
@@ -0,0 +1,119 @@
...
+            createStatement.setString(1, manufacturer.getName());
+            createStatement.setString(2, manufacturer.getCountry());
+            createStatement.executeUpdate();
+            ResultSet generatedKeys = createStatement.getGeneratedKeys();
+            if (generatedKeys.next()) {
+                Long id = generatedKeys.getObject(1, Long.class);
+                manufacturer.setId(id);
+            }
+        } catch (SQLException e) {
+            throw new RuntimeException("Cant insert manufacturer to DB " + manufacturer, e);
```
Comment (Sofasmile) [inline src/main/java/mate/jdbc/dao/ManufacturerDaoImpl.java:34]: https://mate-academy.github.io/jv-program-common-mistakes/java-JDBC/jdbc-intro/JDBC-intro_checklist.html
image

File: src/main/java/mate/jdbc/dao/ManufacturerDaoImpl.java [hunk 10]
```diff
@@ -0,0 +1,119 @@
...
+                        + " SET name = ?, country = ? WHERE id = ? AND id_deleted = false;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement updateStatement =
+                        connection.prepareStatement(updateStatementSql)) {
+            updateStatement.setString(1, manufacturer.getName());
+            updateStatement.setString(2, manufacturer.getCountry());
+            updateStatement.setLong(3, manufacturer.getId());
+            updateStatement.executeUpdate();
+        } catch (SQLException e) {
+            throw new RuntimeException();
```
Comment (Sofasmile) [inline src/main/java/mate/jdbc/dao/ManufacturerDaoImpl.java:88]: let's pass informative message to the exseption

File: src/main/java/mate/jdbc/dao/ManufacturerDaoImpl.java [hunk 11]
```diff
@@ -0,0 +1,119 @@
...
+        }
+        return get(manufacturer.getId()).orElse(null);
```
Comment (Sofasmile) [inline src/main/java/mate/jdbc/dao/ManufacturerDaoImpl.java:90]:
```suggestion
        return manufacturer;
```

File: src/main/java/mate/jdbc/dao/ManufacturerDaoImpl.java [hunk 12]
```diff
@@ -0,0 +1,119 @@
...
+
+    @Override
+    public boolean delete(Long id) {
+        String deleteByIdStatementSql =
+                "UPDATE manufacturers SET is_deleted = true WHERE id = ?;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement deleteByIdStatement =
+                        connection.prepareStatement(deleteByIdStatementSql)) {
+            deleteByIdStatement.setLong(1, id);
+            return deleteByIdStatement.executeUpdate() >= 1;
```
Comment (Sofasmile) [inline src/main/java/mate/jdbc/dao/ManufacturerDaoImpl.java:101]: image
https://mate-academy.github.io/jv-program-common-mistakes/java-JDBC/jdbc-intro/JDBC-intro_checklist.html

File: src/main/java/mate/jdbc/dao/ManufacturerDaoImpl.java [hunk 13]
```diff
@@ -0,0 +1,119 @@
...
+        } catch (SQLException e) {
+            throw new RuntimeException("Cant delete manufacturer by id from DB " + id, e);
+        }
+    }
+
+    private Manufacturer getManufacturer(ResultSet resultSet) {
+        try {
+            String name = resultSet.getString("name");
+            String country = resultSet.getString("country");
+            Long manufacturersId = resultSet.getObject("id", Long.class);
```
Comment (Sofasmile) [inline src/main/java/mate/jdbc/dao/ManufacturerDaoImpl.java:111]:
```suggestion
            Long manufacturerId = resultSet.getObject("id", Long.class);
```

File: src/main/java/mate/jdbc/util/ConnectionUtil.java [hunk 14]
```diff
@@ -0,0 +1,23 @@
...
+
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.SQLException;
+import java.util.Properties;
+
+public class ConnectionUtil {
+    public static Connection getConnection() {
+        try {
+            Class.forName("com.mysql.cj.jdbc.Driver");
```
Comment (Sofasmile) [inline src/main/java/mate/jdbc/util/ConnectionUtil.java:11]: let's move `Class.forName("com.mysql.cj.jdbc.Driver");` to the static block and do it once

**Review by Sofasmile — CHANGES_REQUESTED** [review]
great job)

File: src/main/java/mate/jdbc/dao/ManufacturerDaoImpl.java [hunk 15]
```diff
@@ -67,7 +68,7 @@ public Manufacturer create(Manufacturer manufacturer) {
                 manufacturers.add(getManufacturer(resultSet));
             }
         } catch (SQLException e) {
-            throw new RuntimeException("Cant get manufacturer by id from DB", e);
+            throw new DataProcessingException("Cant get manufacturer by id from DB", e);
```
Comment (Sofasmile) [inline src/main/java/mate/jdbc/dao/ManufacturerDaoImpl.java:71]:
```suggestion
            throw new DataProcessingException("Cant get all manufacturers from DB", e);
```

File: src/main/java/mate/jdbc/dao/ManufacturerDaoImpl.java [hunk 16]
```diff
@@ -98,22 +99,24 @@ public boolean delete(Long id) {
                 PreparedStatement deleteByIdStatement =
                         connection.prepareStatement(deleteByIdStatementSql)) {
             deleteByIdStatement.setLong(1, id);
-            return deleteByIdStatement.executeUpdate() >= 1;
+            int updatedRows = deleteByIdStatement.executeUpdate();
+            return updatedRows >= 1;
```
Comment (Sofasmile) [inline src/main/java/mate/jdbc/dao/ManufacturerDaoImpl.java:103]: image

**Review by Sofasmile — CHANGES_REQUESTED** [review]

**Review by Sofasmile — APPROVED** [review]

**Pull request closed without merging**